src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -514,16 +522,137 @@ teardown_pid1(void)
     reboot(reboot_cmd);
 }
 
+static bool
+sol_platform_linux_micro_should_debug(struct gdbserver *gdb)
+{
+    struct sol_file_reader *fr;
+    struct sol_str_slice file;
+    struct sol_vector tokens;
+    struct sol_str_slice *token;
+    uint16_t i;
+    bool res = false;
+
+    fr = sol_file_reader_open("/proc/cmdline");
+    if (!fr) {
+        SOL_ERR("Could not open /proc/cmdline");
+        return false;
+    }
+
+    file = sol_file_reader_get_all(fr);
+    // remove the \n in the end of the file (this is a single line file - always)
+    file.len--;
+    tokens = sol_util_str_split(file, " ", 0);
+    SOL_VECTOR_FOREACH_IDX (&tokens, token, i) {
+        char arg[PATH_MAX] = { 0 };
+        int err;
+
+        err = snprintf(arg, sizeof(arg), "%.*s", SOL_STR_SLICE_PRINT(*token));
```

Comment:
this is not required, use `sol_str_slice_eq()` (instead of a macro with a string, use a str slice) and instea of `strdup()` use `sol_str_slice_to_string()`.